src/Microsoft.DotNet.ProjectModel/ProjectReader.cs

```diff
@@ -246,6 +243,38 @@ public Project ReadProject(Stream stream, string projectName, string projectPath
             return project;
         }
 
+        internal static string NormalizeProjectDirectoryPath(string path)
+        {
+            if (File.Exists(path) &&
```

Comment:
That's true when you don't have a set of tests and unit tests protecting you from breaking things. Plus, I suggested just a couple of extract methods refactoring to make this method more readable. Just use resharper, mark the code and say extract method. Give it a nice descriptive name and call it a day.